docs/dependencies.md

```diff
@@ -0,0 +1,188 @@
+different index, etc. This metadata must be expressed separately because the `project.dependencies` standard does not allow these common patterns.
+
+## Project dependencies
+
+The `project.dependencies` table represents the dependencies that are used when uploading to PyPI or
+building a wheel. Individual dependencies are specified using [PEP 508](#pep-508) syntax, and the table follows the [PEP 621](https://packaging.python.org/en/latest/specifications/pyproject-toml/)
+standard.
+
+`project.dependencies` defines the packages that are required for the project, along with the version constraints that should be used when installing them.
+
+`project.dependencies` is structured as a list. Each entry includes a dependency name and
+version. An entry may include extras or environment markers for platform-specific packages. For example:
+
+```toml
+[project]
+name = "albatross"
+version = "0.1.0"
+dependencies = [
+  # Any version in this range
+  "tqdm >=4.66.2,<5",
+  # Exactly this version of torch
+  "torch ==2.2.2",
+  # Install transformers with the torch extra
+  "transformers[torch] >=4.39.3,<5",
+  # Only install this package on older python versions
+  # See "Environment Markers" for more information
+  "importlib_metadata >=7.1.0,<8; python_version < '3.10'",
+  "mollymawk ==0.1.0"
+]
+```
+
+If the project only requires packages from PyPI or a single `--index-url`, then `project.dependencies` is sufficient. If, the project depends on local packages, Git dependencies, or packages from a different
+index, `tool.uv.sources` will be needed.
+
+## Dependency sources
+
+During development, the project may rely on a package that isn't available on PyPI. In the following example, the project will require several package sources:
+
+- `tqdm` from a specific Git commit
+- `importlib_metadata` from a dedicated URL
+- `torch` from the PyTorch-specific index
+- `mollymawk` from the current workspace
+
+These requirements can be expressed by extending the definitions in the `project.dependencies` table with `tool.uv.sources` entries:
+
+```toml
+[project]
+name = "albatross"
+version = "0.1.0"
+dependencies = [
+  # Any version in this range.
+  "tqdm >=4.66.2,<5",
+  # Exactly this version of torch.
+  "torch ==2.2.2",
+  # Install transformers with the torch extra.
+  "transformers[torch] >=4.39.3,<5",
+  # Only install this package on Python versions prior to 3.10.
+  "importlib_metadata >=7.1.0,<8; python_version < '3.10'",
+  "mollymawk ==0.1.0"
+]
+
+[tool.uv.sources]
+# Install a specific Git commit.
+tqdm = { git = "https://github.com/tqdm/tqdm", rev = "cc372d09dcd5a5eabdc6ed4cf365bdb0be004d44" }
+# Install a remote source distribution (`.zip`, `.tar.gz`) or wheel (`.whl`).
+importlib_metadata = { url = "https://github.com/python/importlib_metadata/archive/refs/tags/v7.1.0.zip" }
+# Pin a dependency for a specific registry.
+torch = { index = "torch-cu118" }
```

Comment:
Has the index source already been supported?? It seems to be missing from the list of sources supported by `uv` below.

Reply:
The doc was originally written assuming support but we've cut it from the scope of the initial stable release. I'll remove here, thanks!